src/Symfony/Bundle/FrameworkBundle/Error/PagesDumper.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Error;
+
+use Symfony\Bridge\Twig\ErrorRenderer\TwigErrorRenderer;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\WebpackEncoreBundle\Asset\EntrypointLookupInterface;
+
+/**
+ * @author Loïck Piera <pyrech@gmail.com>
+ *
+ * @final
+ */
+class PagesDumper
+{
+    public function __construct(
+        private string $kernelCacheDir,
+        private Filesystem $filesystem,
+        private ?TwigErrorRenderer $twigErrorRenderer = null,
+        private ?EntrypointLookupInterface $entrypointLookup = null,
+    ) {
+    }
+
+    public function dump(?string $targetDirectory = null): void
+    {
+        if (null === $this->twigErrorRenderer) {
+            throw new \LogicException('You cannot dump error pages if the Twig Bundle is not available. Try running "composer require symfony/twig-bundle".');
+        }
+
+        $targetDirectory ??= $this->kernelCacheDir.\DIRECTORY_SEPARATOR.'error_pages';
+        $errorStatusCodes = array_filter(
+            array_keys(Response::$statusTexts),
+            fn ($statusCode) => $statusCode >= 400
+        );
+
+        foreach ($errorStatusCodes as $errorStatusCode) {
```

Comment:
The list of status codes should be configurable to allow users to choose which error they want to suppress.